test_pull_requests

```diff
@@ -394,63 +393,16 @@ class FlushableStore
     @cache.delete(key)
   end
 
-  # Iterate and deserialize across a response entry to extract the URL
-  # being cached for.
-  def get_entry_with_url(serialized_entries)
-    entry = nil
-    serialized_entries.each do |serialized_entry|
-      serialized_entry.each do |sub_entry|
-        entry = Marshal.load sub_entry
-        break if entry.include?(:method) && entry.include?(:url)
-      end
-    end
-    entry
-  end
-
-  # GitHub-specific (although probably other APIs are similar) method
-  # to remove the pagination parameters from a URL so it can be better
-  # classified
-  #
-  # Example:
-  #   synthesize_url("https://api.github.com/some/resource/id?per_page=100")
-  #     returns
-  #   "https://api.github.com/some/resource/id"
-  def synthesize_url(orig_url)
-    url = orig_url.dup
-    uri = Addressable::URI.parse url
-    params = uri.query_values
-    if params && params.include?('per_page')
-      params.delete 'per_page'
-      uri.query_values = params
-      url = uri.to_s.chomp('?')
-    end
-    url
-  end
-
-  # check for cache entries with URLs matching the provided URL, and
-  # remove them from the cache
-  def invalidate_url(orig_url)
-    url = synthesize_url(orig_url)
-    invalidate = @invalidatable.delete(url.to_s)
-    @cache.delete_if { |k,v| invalidate.include? k } if invalidate
+  def write(key, value)
+    @cache[key] = value
   end
 
-  # On write, classify the synthesized URL for later invalidation
-  def update_invalidatable(key, serialized_entries)
-    entry = get_entry_with_url(serialized_entries)
-    url = synthesize_url(entry[:url])
-    if [:get, :head].include? entry[:method]
-      @invalidatable[url][key] = true
-    elsif [:post, :put, :delete, :patch].include? entry[:method]
-      invalidate_url(url)
+  def invalidate_urls(url=nil)
+    if @storage and @storage.respond_to? :invalidate_urls
+      @storage.invalidate_urls(url)
```

Comment:
@stevekuznetsov counter-nit: the method takes in a single url, but may invalidate multiple urls loosely matching the provided url.